### ISSUE TITLE
HOW-TO configure for my own SSL Cert

### ISSUE THREAD

**Issue by darue-keller** [skavanagh/KeyBox#129]
It's looking like I'm going to have to import the cert-chain into the existing etc/keystore file (I actually already have this in a keystore file of my own). I'm not seeing what causes the program to pick one cert-chain over another, is it by alias? In order to investigate further I'll need to find out how to UN-obfuscate the default keystore file's password, or someone could just tell me what it is.

**Comment by skavanagh** [comment]
It just uses jetty and they have documentation on their site

https://wiki.eclipse.org/Jetty/Howto/Configure_SSL

You should just be able to make your own keystore/truststore

> keytool -keystore keystore -import -alias jetty -file cert.crt

and import your certs (just don't run that on top of the existing keystore).  If there isn't a keystore there it will create one with the password you set.

then just replace the keystore in the jetty/etc/ directory and you set the passwords in the jetty/modules/ssl.mod file

see

http://www.eclipse.org/jetty/documentation/current/configuring-security-secure-passwords.html

on how to get the password format.

**Comment by darue-keller** [comment]
thanks much, I've got my own keystore file in place now and things are working!
one note: I changed the password in jetty-ssl-context.xml rather than the ssl.mod file. I didn't even change the stuff in the ssl.mod file, yet it's working.

There are three places in `jetty-ssl-context.xml` where a password is stored:
```
KeyStorePassword
KeyManagerPassword
TrustStorePassword
```
I understand the KeyStore and password fields, but the TrustStore (which seems to be the same file in this config) I'm not 100% clear on, I'm guessing that would be for storing client-certificates. but what would the KeyManager thing be used for?
